make column names case sensitive

awclives opened the pull request:
This bit me on my first usage of tablesaw.

I think we also should enforce uniqueness of column names in addColumn.  I think it should throw an exception if the name already exists.  Uniqueness is assumed and required in many places so I think it should be enforced.

lwhite1 commented:
I agree that enforcing column name uniqueness is important. I will add an issue to track it until it's fixed.
But I'm not sure I agree about making the names case-sensitive. How did this bite you?

My feeling is that the user will be often working with files created by others, with arbitrary column name capitalization and that it will impose an avoidable cognitive task to remember the capitalization of each name. 

How would you feel about just ensuring case-insensitive uniqueness?

awclives commented:
case-insensitive uniqueness sounds good to me.

that would have defanged the issue that bit me.  i added "Title" and "title" columns just as a little test of the library, and then figured out that the resulting CSV was screwed up due to the non-unique columns.  if it had thrown an exception when I added the second one, as you are proposing, I would have been happy.

lwhite1 commented:
I will make it so :)

lwhite1 commented:
closing after making changes to enforce column name uniqueness
